src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/CommandLineItemHandler.cs

```diff
@@ -66,11 +70,27 @@ private static bool ProcessDesignTimeBuildFailure(IProjectChangeDescription proj
             return designTimeBuildFailed;
         }
 
-        private static void ProcessOptions(IProjectChangeDescription projectChange, IWorkspaceProjectContext context)
+        private void ProcessOptions(IProjectChangeDescription projectChange, IWorkspaceProjectContext context)
         {
             // We don't pass differences to Roslyn for options, we just pass them all
             IEnumerable<string> commandlineArguments = projectChange.After.Items.Keys;
             context.SetOptions(string.Join(" ", commandlineArguments));
+
+            // The rule set file is specified as an option in the command line arguments, but it is reduced
+            // down to a map from diagnostic ID to severity as part of processing those options.
+            // The language service also needs to know about the file, separate from the options, so that it
+            // can watch it for changes.
+            string ruleSetOption = projectChange.After.Items.Keys.SingleOrDefault(option => option.StartsWith("/ruleset:"));
+            if (ruleSetOption != null)
+            {
+                string ruleSetPath = ruleSetOption.Substring("/ruleset:".Length);
+                string rootedRuleSetPath = _project.MakeRooted(ruleSetPath);
+                context.SetRuleSetFile(rootedRuleSetPath);
+            }
+            else
+            {
+                context.SetRuleSetFile(string.Empty);
```

Comment:
@mavasani `projectChange.After` must have the complete set of options, so if there's not /ruleset option, then we need to unset the rule set in the language service. Note that the `SetRuleSetFile` implementation on the LS side already bails out early if the new value and existing value are the same.